Update README.md (#184)
gammapy/gammapy-meetings: dev-meetings/2025/2025-07-18/README.md

  - Chat with ??? nowfocus (???), could be useful for gammapy.
 
RT: xarray possible replacement for maps at some point?
    Regarding zfit, overview important for possible restructuring at some point (?)
    AD: zfit using not an option, but can exchange design ideas / learn from each other.
  
### Documentation
* general updates made to tutorials to be formatted correctly
* added documentation on skipping github actions to the dev how to
* reordering of the tutorials

KF reports on the status. Especially:
- VERITAS tutorial [#5918](https://github.com/gammapy/gammapy/pull/5918)
KF: VERTIAS tutorial basically ready.
RT: VERITAS turotial should be good to go, AS agrees, one more approval needed only, RT agrees to approve PR, let AS merge.
Maybe safer to use regular safe-regions?

* to add descriptions to the various `peek` functions -- delegation of tasks [#5961](https://github.com/gammapy/gammapy/pull/5961)
  - PR started for improving peek-functions, delegation of tasks needed.

### Outstanding:

AS: Can we talk about #5979, #5982 ?

* [#5979](https://github.com/gammapy/gammapy/pull/5979) Fix memory leak related to lru_cache - Quentin Remy

AD comments:

> There was a memory leak, long standing issue. lru cach is dict with keys are arguments passed to a func, value is the computed, returned value by this function. Problem with implementation is: caches is global one, keeps reference to the class object itself. If geometries out of scope, caching still keeps reference to geometry object, not garbage collected -> mem leak. Solution: Weak reference for geometry objects. 

Reviewer: ```suggestion
There was a memory leak, long standing issue. `lru_cache` wrapper applied to a function creates a dict which associates arguments passed to the function to the value it returns. Problem with implementation is: caches is global one, keeps reference to the class object itself. If geometries go out of scope, caching still keeps a reference to the geometry object, hence they are not garbage collected and memory leak ensues. Solution: use weak reference for geometry objects. 
```

RT: If introduce equality condition -> tolerance condition. Is this in general completely safe? X-Ray data for example, we need to check.
AD: Commented on this in PR, good way to define meaningful hash of geom-object.
AS: What do you mean by that?
AD: Clarifies. "__" returns unique identifier by object, that is defined by property on the object.
What are the properties of a geom-object, that make it equal (???) to other one? Not trivial.
In current state, PR maybe to risky. Two things needed: Hashing of geom object and if definable in meaningful way, second: real world impact?

* tutorial on upper limits [#5933](https://github.com/gammapy/gammapy/pull/5933)
* tutorial for non-detection [#5969](https://github.com/gammapy/gammapy/pull/5969)

RT: What is the status?
  AS: Was maybe discussed last week?
  AS recapitulates the status.
On #5969:
AS recapitulates RTs comments: Two different things: Making PSMap or Fluxmap, or searching for source at specific point.
RT: Use case is, looking for a source. If it is there, have UL on total flux. But so far, specific pos given.
Focus on question of known/expected source at known position. Is the object emitting in my data, or not.

> Discussion between AS and RT.

Reviewer: ```suggestion
```

*need to make a decision on the tutorial ordering adaptions [#5949](https://github.com/gammapy/gammapy/pull/5949)

RT: Make API parts more visible, too many things in the API-part. Discussing of "key-steps" part, clarification needed, not everything as key-step.
AS: astro_dark_matter also in recipes. Put it only once? Uncomfortable with name "key-steps", suggests you have to do it first.
KF answers: You can argue, that one should understand the points beforehand. If looking for having extra section for example, things like clustering, theta_square_plot.py would be moved there. Question: What do we want to change for 2.0, if anything.
RT: Maybe keep API part for DM, priors, etc and something specific for smaller use-cases.
Decision regarding astro_dark_matter.py not taken yet. 

> KF: Leaving it in the turotials makes more sence yet, not clear how many people go to the recipes.

Reviewer: ```suggestion
KF: Leaving it in the turotials makes more sense yet, not clear how many people go to the recipes.
```

RT: We can not rely on gammapy-recipes, as long as it is not stable.
KF, RT agrees that specific opinions could be put on the PR.
RT: Decision taking needed soon.

* can we merge? [#5886](https://github.com/gammapy/gammapy/pull/5886/)
* command line option for tutorial setup -- should we remove from all other tutorials now too? [#5941](https://github.com/gammapy/gammapy/pull/5941/)

### PRs and Timeline:
RT: Many open PR yet with 2.0-milestone. Timeline? End of July not realistic.

> We need to branch, create 2.0, test it. Who is around?

Reviewer: ```suggestion
We need to branch, create 2.0rc, test it. Who is around?
```

->AS,KF,BK, available probably in the coming time.
Open point, changelog needs to be prepared, AS agrees to do it.

> Citations takes time as well (especially. people from last coding sprint).

Reviewer: ```suggestion
Citation.cff update takes time as well (especially miss input from people from last coding sprint).
```

* Discussion of PR #5881 "A class for vectorized evaluation of flux uncertainties based on samples". 

> RT reports: Inconsistency how evaluation is done e.g. on naima models and ??? models, not uniform yet.

Reviewer: ```suggestion
RT reports: Inconsistency how evaluation is done e.g. on naima models and compounds models, not uniform yet.
```

Evaluate function must be vectorizable, otherwise errorband can not be plotted. Do we need fall-back-mechanism?
For-loop eval of all the parameters?

> Discussion with AS.

Reviewer: ```suggestion
```

RT: Main thing that needs to be done: Introduce fall back mechanism.
Regarding whole PR, feature really needed, PR should go in for 2.0.

* Brief discussion of PR #5838 "Addition of FitStatistic penalties"

> RT: Can come later.

Reviewer: ```suggestion
RT: Can come later. But can go undocumented too...
```

* Brief discussion of PR #5815

>  RT: Have to talk with BK.

Reviewer: ```suggestion
 RT: Have to talk with BK about scope of function.
```

RT: Regarding code, do we agree to postpone to mid august / Aug 20? 
AS agrees, end of August should be hard deadline.
RT: Branching maybe in two weeks.

* Brief discussion of PR #5910 with KF.

  KF: First changed only name of params, but tests fails. What is final verdict?

>   RT: Have to check how it is used in the code.

Reviewer: ```suggestion
  RT: Have to check how it is used in the code and make sure it is consistent
```

* Brief discussion of PR #5960. 

> AS: Discussed last week. Issue that edges are in true_energy, but cut in recon energies (???).

Reviewer: ```suggestion
AS: Discussed last week. Issue that edges are in true_energy, but cut in reco energy.
```

KF agrees. How to do it?

> RT: Not go in direction of weights, rather tresholds. It would otherwise change the logic. You can try to see, if it is possible, but would be different mechanism.

Reviewer: ```suggestion
RT: Not go in direction of weights, rather hard thresholds. It would otherwise change the logic. You can also try to see if it is possible to filter on reco exposure instead, but would be different mechanism.
```
